src/test/java/se/liu/ida/hefquin/engine/data/utils/SolutionMappingsIterableWithOneVarFilterTest.java

```diff
@@ -0,0 +1,80 @@
+package se.liu.ida.hefquin.engine.data.utils;
+
+import org.apache.jena.graph.Node;
+import org.apache.jena.graph.NodeFactory;
+import org.apache.jena.sparql.core.Var;
+import org.apache.jena.sparql.engine.binding.Binding;
+import org.junit.Test;
+import se.liu.ida.hefquin.engine.data.SolutionMapping;
+import se.liu.ida.hefquin.engine.data.impl.SolutionMappingUtils;
+import se.liu.ida.hefquin.engine.datastructures.impl.SolutionMappingsHashTableBasedOnOneVar;
+import se.liu.ida.hefquin.engine.datastructures.impl.SolutionMappingsIndexBase;
+
+import java.util.*;
+
+import static org.junit.Assert.*;
+
+public class SolutionMappingsIterableWithOneVarFilterTest {
+    @Test
+    public void hashTableWithOneInputVariable() {
```

Comment:
Why is this test called something with "hashTable"? Hash tables are not relevant for the these helper classes.

Similarly, in many of the other test class for these classes you have "hashTable..." tests.